Device class does not instantiate properly

## TL;DR

Cannot initiate page object if `_config.properties` file does not specify `capabilities.platfrom`/`capabilities.platformName`

## Current situation

When initiate a Page object using `initPage(<class>)` method, the current `Device` object is used to retrieve the device os:

```
Type screenType = IDriverPool.getDefaultDevice().getDeviceType();
```

From that info, appropriate concreate class (iOS/Android) is instantiated.

The `Device`'s OS information is set in its constructor using `Configuration`:

```
setOs(Configuration.getPlatform());
```

The `Configuration` will in turn look for `platfrom` in `R.CONFIG`, which can be modify using `_config.properties` file.

## Problem

The problem is when we have very lean `_config.properties` file (without `platform` information), and create the driver using explicit capabilities:

```
getDriver(DEFAULT, caps, "http://localhost:4444/wd/hub");
```

The default `Device` object will get invalid OS information, making it impossible to use `initPage`.

```
Caused by: java.lang.RuntimeException: Incorrect driver type. Please, check config file for name: Default_iOS_device; type: phone; os: *; osVersion: 11.4; udid: 5e394a7d522603fe82315083f825f6311ffdfe02; remoteURL: null
        at com.qaprosoft.carina.core.foundation.webdriver.device.Device.getDeviceType(Device.java:173)
        at com.qaprosoft.carina.core.foundation.utils.factory.ICustomTypePageFactory.initPage(ICustomTypePageFactory.java:51)
        at com.qaprosoft.carina.core.foundation.utils.factory.ICustomTypePageFactory.initPage(ICustomTypePageFactory.java:38)
        at vn.zalopay.client.cucumber.steps.BaseSteps.<init>(BaseSteps.java:18)
        at vn.zalopay.client.cucumber.steps.TypeRegistry.<init>(TypeRegistry.java:15)
        ... 44 more
```

--- a/carina-webdriver/src/main/java/com/qaprosoft/carina/core/foundation/webdriver/device/Device.java
+++ b/carina-webdriver/src/main/java/com/qaprosoft/carina/core/foundation/webdriver/device/Device.java
@@ -29,6 +29,7 @@ import org.apache.commons.lang3.StringUtils;
 import org.apache.log4j.Logger;
 import org.openqa.selenium.Capabilities;
 import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.remote.DesiredCapabilities;
 
 import com.qaprosoft.carina.commons.models.RemoteDevice;
 import com.qaprosoft.carina.core.foundation.commons.SpecialKeywords;
@@ -106,7 +107,7 @@ public class Device extends RemoteDevice implements IDriverPool {
         }
         setType(deviceType);
 
-        setOs(Configuration.getPlatform());
+        setOs(Configuration.getPlatform(new DesiredCapabilities(capabilities)));
 
         String platformVersion = R.CONFIG.get(SpecialKeywords.MOBILE_DEVICE_PLATFORM_VERSION);
         if (!R.CONFIG.get(SpecialKeywords.MOBILE_DEVICE_BROWSERSTACK_PLATFORM_VERSION).isEmpty()) {

